src/adapter/src/catalog.rs

```diff
@@ -703,7 +701,12 @@ impl CatalogState {
             role_id: MZ_SYSTEM_ROLE_ID,
             prepared_statements: None,
         };
-        enable_features_required_for_catalog_open(&mut session_catalog);
+
+        // Enable catalog features that might be required during planning in
+        // [Catalog::open]. Existing catalog items might have been created while a
+        // specific feature flag turned on, so we need to ensure that this is also the
+        // case during catalog rehydration in order to avoid panics.
+        session_catalog.system_vars_mut().enable_all_feature_flags();
```

Comment:
> The suggestion to enable all feature flags by default is the equivalent to how `--unsafe-mode` currently works and is behavior I believe we should move away from. `--unafe-mode` has the bad behavior that CI runs do not reflect the experience of how production behaves, and those differences should be opt-in rather than on-by-default. In the past, we've run into panics in production explicitly because CI's behavior allowed something that prod did not.
> 
> The majority of the feature flags are meant to gate behavior in SQL planning, which just prevents certain code paths from being hit by customers. This does not reconfigure default values for e.g. persist config, which looks like is how pubsub is enabled.
> 
> If there are more or better docs you'd like to see, I'm very open to suggestions to improve the guidance here.

Strong agree!